src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -1360,6 +1360,20 @@ public LdapUserDetails loadUserByUsername(String username) throws UsernameNotFou
                             user.addAuthority(extraAuthority);
                         }
                     }
+
+                    if (!UserAttributesHelper.checkIfAccountNonExpired(v)) {
+                        user.setAccountNonExpired(false);
+                    }
+                    if (!UserAttributesHelper.checkIfAccountNonLocked(v)) {
+                        user.setAccountNonLocked(false);
+                    }
+                    if (!UserAttributesHelper.checkIfCredentialsAreNonExpired(v)) {
+                        user.setCredentialsNonExpired(false);
+                    }
+                    if (!UserAttributesHelper.checkIfUserIsEnabled(v)) {
+                        user.setEnabled(false);
+                    }
```

Comment:
Could these be simplified to something like
user.setEnabled(UserAttributesHelper.isUserEnabled())
?

Reply:
Yes, good point!